pkg/runtime/runtime.go

```diff
@@ -3079,11 +3079,12 @@ func (a *DaprRuntime) startSubscriptions() {
 
 // Stop subscriptions to all topics and cleans the cached topics
 func (a *DaprRuntime) stopSubscriptions() {
-	// Stop all subscriptions by canceling the subscription context
+	if a.pubsubCtx == nil || a.pubsubCtx.Err() != nil { // no pubsub to stop
```

Comment:
Sadly this method is not concurrency-safe. If 2 goroutines call `stopSubscriptions` at the same time, one could set `pubsubCancel = nil` while the other one is trying to invoke `a.pubsubCancel`

I think it's fine to just call `pubsubCancel` multiple times and not set it to nil.

Reply:
Good point. The same issue as to setting pubsubCtx to nil also exists here.

Reply:
@ItalyPaleAle @mukundansundar - if this was a possibility with pubsubCancel, would go test -race not flag this?

Reply:
Only if a test was built around those methods